tools/resourcedocsgen/cmd/docs/registry.go

```diff
@@ -39,6 +40,76 @@ import (
 	concpool "github.com/sourcegraph/conc/pool"
 )
 
+// PackageMetadataProvider is an interface for providers that retrieve package metadata
+// from either the filesystem directory or the Pulumi Registry API.
+type PackageMetadataProvider interface {
+	// GetPackageMetadata returns metadata for a specific package
+	GetPackageMetadata(pkgName string) (*pkg.PackageMeta, error)
+	// ListPackageMetadata returns metadata for all packages
+	ListPackageMetadata() ([]*pkg.PackageMeta, error)
```

Comment:
Unless these can be nil (with a nil error), can we just return the structs directly?
```suggestion
	GetPackageMetadata(pkgName string) (pkg.PackageMeta, error)
	// ListPackageMetadata returns metadata for all packages
	ListPackageMetadata() ([]pkg.PackageMeta, error)
```